File: 프로그래머스/0/181913. 문자열 여러 번 뒤집기/문자열 여러 번 뒤집기.js
function solution(my_string, queries) {
    const arr = my_string.split('');
    
    queries.forEach(([s, e]) => {
        const r = arr.slice(s, e+1).reverse();
        arr.splice(s, e-s+1, ...r);
    })
    
    return arr.join('');
}

File: 프로그래머스/0/181913. 문자열 여러 번 뒤집기/문자열 여러 번 뒤집기.java
import java.util.*;

class Solution {
    public String solution(String my_string, int[][] queries) {
        StringBuilder sb = new StringBuilder(my_string);
        
        for (int[] query : queries) {
            String reversed = new StringBuilder(sb.substring(query[0], query[1] + 1))
                .reverse()
                .toString();
            sb.replace(query[0], query[1] + 1, reversed);
        }
        
        return sb.toString();
    }
}